### PR TITLE
[RigidSpaces] Make initial definitions and state lemmas

### PR DESCRIPTION
**Pull request by jcreedcmu**
Added some thoughts about [rigid spaces](https://jcreedcmu.github.io/posts/2023-01-14/).
Feel free to decline if you feel it isn't "on topic" for this repo, but perhaps it could be another fun formalization puzzle.

### CODE REVIEW

**Review by dwrensha — COMMENTED** [review]
When you add a new file like this, you should add it to `MathPuzzles.lean` so that it gets picked up by `lake build` (and CI).

Do you anticipate adding more theory building upon this (below the TODO theorems at the bottom)? If so, then this stuff might better fit in its own repository.

File: MathPuzzles/RigidSpaces.lean [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+-- https://jcreedcmu.github.io/posts/2023-01-14/
+
+-- I'm not completely sure about what the ideal approach is with
+-- respect to lean type classes vs. structures for defining
+-- sets-with-structure in this setting.
+
+-- FIXME: The current definition below probably isn't quite right,
+-- because the topology on E is not constrained in any way to reflect
+-- the vector space structure on E.  I intend to assume E to be a
+-- finite-dimensional real vector space, with the usual topology on
```
[the remark this answers is not in the document]
Comment (dwrensha) [reply MathPuzzles/RigidSpaces.lean:16]: Ah, I suppose it's just: any choice of inner product yields an equivalent topology. I wonder if that's in mathlib somewhere. If not, that could be a fun exercise.

[the remark this answers is not in the document]
Comment (dwrensha) [reply MathPuzzles/RigidSpaces.lean:16]: Ah, apparently that only works in the finite-dimensional case. See https://en.wikipedia.org/wiki/Box_topology .

I don't see any assumptions in your code that forces `E` to have finite dimension. Is that going to be a problem?

**Comment by jcreedcmu** [comment]
Yeah, maybe if this is an open-ended formalization rather than a closed-off puzzle then maybe this is not the right repo for it.

**Pull request closed without merging**